pytorch_lightning/trainer/trainer.py

```diff
@@ -1046,12 +1051,13 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         self._pre_dispatch()
 
         if self.accelerator.restore_checkpoint_after_pre_dispatch:
-            if self._ckpt_path:
-                self._load_checkpoint_weights()
-            self._restore_modules_and_callbacks()
+            self._restore_modules_and_callbacks(ckpt_path)
 
         # restore optimizers, etc.
-        self.checkpoint_connector.restore_training_state()
+        if self.state.fn == TrainerFn.FITTING:
+            self.checkpoint_connector.restore_training_state()
+
+        self.checkpoint_connector.resume_end()
```

Comment:
n00b question: why is this bumped up to here vs in `_pre_training_routine` ?

Reply:
now that this calls `resume_start` for all (not only fitting), similarly `resume_end` for all as well